lens/interface.go

```diff
@@ -0,0 +1,25 @@
+package lens
+
+import (
+	"context"
+	"github.com/filecoin-project/go-address"
+	api "github.com/filecoin-project/lotus/api"
+	"github.com/filecoin-project/lotus/chain/types"
+	"github.com/filecoin-project/specs-actors/actors/util/adt"
+	"github.com/ipfs/go-cid"
+)
+
+type LensAPI interface {
```

Comment:
Agreed, the client-side interface pattern described looks to be more maintainable. Will modify this to instead be:
```go
type LensAPI interface {
  node.FuullNode
  Store() adt.Store
}
```
and pass that to methods as is being done now. This will allow methods that use the `adt.Store` to take advantage of the [caching](https://github.com/filecoin-project/sentinel-visor/pull/10/files#diff-34be193850e3eb83f186727d49339b50R56).